File: persons-finder-projects/java/persons-finder-4j/src/main/java/com/persons/finder/repositories/LocationRepository.java
/**
 * 
 */
package com.persons.finder.repositories;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;

import com.persons.finder.data.Location;

/**
 * 
 */
public interface LocationRepository 
	extends JpaRepository<Location, Long> {
	
	Optional<Location> findByReferenceId(Long id);
	Optional<Location> findByLatitudeAndLongitude(Double latitude, Double longitude);
}

File: persons-finder-projects/kotlin/persons-finder/src/main/kotlin/com/persons/finder/repositories/LocationRepository.kt
package com.persons.finder.repositories

import com.persons.finder.data.Location
import org.springframework.data.repository.CrudRepository
import java.util.Optional

interface LocationRepository : CrudRepository<Location, Long> {
    fun findByReferenceId(id: Long): Optional<Location>
    fun findByLatitudeAndLongitude(latitude: Double, longitude: Double): Optional<Location>
}